src/Haskell/Ide/Engine/Options.hs

```diff
@@ -1,7 +1,21 @@
 {-# LANGUAGE CPP #-}
 module Haskell.Ide.Engine.Options where
 
+#if __GLASGOW_HASKELL__ < 804
+import           Data.Semigroup
+#endif
+#if __GLASGOW_HASKELL__ < 808
+import           Data.Monoid                           ((<>))
+#endif
```

Comment:
We could move these to `Compat.hs` but then we would need to import that everywhere else. Not sure if this is better.
Btw, we are not supporting Versions lower than 8.4.2, so I think the first CPP block can be omitted.